exonum/src/storage/proof_map_index/key.rs

```diff
@@ -311,6 +415,111 @@ impl StorageKey for ProofPath {
     }
 }
 
+impl PartialOrd for ProofPath {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        if self.start() != other.start() {
+            return None;
+        }
+        // NB: This check can be moved to "real" code; the code below does not work
+        // if `self.start() % 8 != 0` without additional modifications.
+        debug_assert_eq!(self.start(), 0);
```

Comment:
Personally, I'm against debug assertions for such cases. I prefer to replace it with `assert_eq`. As for me, debug assertions are needed when such checks are costly in terms of performance.